cpp/ql/src/Likely Bugs/Conversion/ArrayArgSizeMismatch.ql

```diff
@@ -17,6 +17,7 @@ where f = c.getTarget() and
         b = paramType.getArraySize() and
         argType.getBaseType().getSize() = paramType.getBaseType().getSize() and
         a < b and
+        a > 0 and
```

Comment:
Don't we have a more sophisticated way to detect pre-C99 encodings of flexible array members? I don't think it's very long ago that we improved this detection for the sake of some other queries.

Reply:
The alternative formulation would be `not memberMayBeVarSize(_, c.getArgument(i).(VariableAccess).getTarget())`.  I actually tried this but got nervous because it assumes `c.getArgument(i)` is a `VariableAccess`.  Perhaps that's a reasonable assumption though - I can't think of a common case where it wouldn't be.

Reply:
I can't think of a case at all where an expression could denote a flexible array member without being a variable access. More specifically, it should always be a `FieldAccess`.